Tests/SystemInfo/Collector/ComposerSystemInfoCollectorTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * File containing the ComposerSystemInfoCollectorTest class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzSupportToolsBundle\Tests\SystemInfo\Collector;
+
+use EzSystems\EzSupportToolsBundle\SystemInfo\Collector\ComposerSystemInfoCollector;
+use EzSystems\EzSupportToolsBundle\SystemInfo\Value\ComposerSystemInfo;
+use PHPUnit_Framework_TestCase;
+
+class ComposerSystemInfoCollectorTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * @var ComposerSystemInfoCollector
+     */
+    private $composerCollector;
+
+    public function setUp()
+    {
+        $this->composerCollector = new ComposerSystemInfoCollector(__DIR__ . '/_fixtures/');
```

Comment:
If you wanna avoid flat files, I have used https://github.com/mikey179/vfsStream in the past. It will let you create a virtual FS (using streams). You would pass it as an argument to the collector, and add a `composer.lock` file from it. Not sure if better, but it is an option, and it doesn't rely on external files.